### PR TITLE
Feature/#2 connect to nexus api

### PR DESCRIPTION
**Pull request by muri-softwareag**
The merge request includes implementation of:

- Rest Template configuration with certificate based authentication.
- A generic rest service to make rest calls to nexus api.
- A sample get request from nexus api.

### CODE REVIEW

File: pkiintegration/src/main/java/com/cumulocity/pkiintegration/rest/CertificateController.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/certificates")
+public class CertificateController {
+	
+	private final CertificateManagementService certificateManagementService;
+	
+	@GetMapping("/registrations")
+	public ResponseEntity<?> getCertificateRegsitrations() throws RestClientException, URISyntaxException {
+		return ResponseEntity.status(HttpStatus.OK).body(certificateManagementService.getRegistrations());
```
[the remark this answers is not in the document]
Comment (muri-softwareag) [reply pkiintegration/src/main/java/com/cumulocity/pkiintegration/rest/CertificateController.java:25]: Ivaylo this is just an endpoint that I created only for testing purpose. I will eventually remove this.

File: pkiintegration/src/main/java/com/cumulocity/pkiintegration/configuration/RestTemplateConfig.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+	@Autowired
+	private ResourceLoader resourceLoader;
+
+	@Bean
+	public RestTemplate restTemplate() throws KeyStoreException, NoSuchAlgorithmException, CertificateException,
+			FileNotFoundException, IOException, UnrecoverableKeyException, KeyManagementException {
+		log.info("Initializing RestTemplate to connect with Certificate-Authority API");
+		String certificatePassword = "FU63C84I";
+		KeyStore clientStore = KeyStore.getInstance("PKCS12");
```
[the remark this answers is not in the document]
Comment (muri-softwareag) [reply pkiintegration/src/main/java/com/cumulocity/pkiintegration/configuration/RestTemplateConfig.java:41]: this is done.,

File: pkiintegration/src/main/java/com/cumulocity/pkiintegration/service/HttpRestRequestService.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class HttpRestRequestService {
+	
+	private static final String nexusBaseUrl = "https://pgw.38.qa.go.nexusgroup.com/api";
```
[the remark this answers is not in the document]
Comment (muri-softwareag) [reply pkiintegration/src/main/java/com/cumulocity/pkiintegration/service/HttpRestRequestService.java]: I've added todo comments.. to move the constants in tenant options. maybe later we can implement that when we are deep in development. At the moment we are just figuring out what and how to do things.

**Comment by kreinhar** [comment]
build rest connector which allows as to create the REST requests

**Pull request closed without merging**